functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py

```diff
@@ -366,16 +366,40 @@ def test011_restart_vm_after_migration(self):
 
         self.lg('%s ENDED' % self._testID)
 
-    @unittest.skip('Not Implemented')
     def test012_check_cloned_vm(self):
-        """ OVC-000
+        """ OVC-029
         *Test case for checking cloned VM ip, portforwards and credentials*
-
         **Test Scenario:**
-
-        #. Create (VM1).
+        #. Create (VM1), should succeed.
+        #. Create portforward to ssh port for (VM1).
```

Comment:
Please take a snapshot for vm1 then write a file on the machine and after the clone make sure this file is there and that u can't rollback using the parent vm snapshot

Reply:
done